### ISSUE TITLE
SSH-remote not accepting password (+OTP)

### ISSUE THREAD

**Issue by ayspunde** [microsoft/vscode-remote-release#6449]
I'm using VS code to connect to the Remote server, the option to enter the password from the VS code is not working. 
The workaround is to run ssh from the command line (iTerm), after that connection from VS Code works.

Steps to Reproduce:
1. Try to connect to the remote server using VS Code with the ssh config like this (jumping to a cluster node through the load balancer):
Host *
    ControlMaster auto
    ControlPath  ~/.ssh/sockets/%r@%h-%p
    ControlPersist  600
    IdentityFile ~/.ssh/nersc
    IdentitiesOnly yes
    ServerAliveCountMax 30
    ServerAliveInterval 30
    ForwardX11 yes
    ForwardX11Trusted yes
    User qwerty
Host nid12345
    ProxyCommand ssh -A -J cori,cmom05 %h
2. When prompted for a password, enter a password, hit enter, repeat...
- no connection

For VS Code Remote to work, I need to connect to the host from the cmd (iTerm):
$> ssh -A -J cori,cmom05 nid12345
(that will ask for the password, then connect)
After cmd ssh worked, SSH-Remote will connect to the remote host without asking for the password.

Basically, ssh password works in the terminal, but not in SSH-Remote.
Local: macOS
Remote: Linux

**Comment by roblourens** [comment]
Please share the log from the Remote-SSH output channel

**Comment by ayspunde** [comment]
[15:34:33.767] Running script with connection command: ssh -T -D 54658 -o ConnectTimeout=60 nid00306 bash
[15:34:34.674] > (ayspunde@cmom05.nersc.gov) Password: 
[15:34:34.674] Got some output, clearing connection timeout
[15:34:34.675] Showing password prompt
[15:35:34.611] "install" terminal command done
[15:35:34.614] Install terminal quit with output: (ayspunde@cmom05.nersc.gov) Password: 
[15:35:34.614] Received install output: (ayspunde@cmom05.nersc.gov) Password: 
[15:35:34.616] Failed to parse remote port from server output
[15:35:34.617] Resolver error: Error: 
	at Function.Create (/Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:586041)
	at Object.t.handleInstallOutput (/Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:584693)
	at Object.t.tryInstall (/Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:681846)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async /Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:644486
	at async Object.t.withShowDetailsEvent (/Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:647831)
	at async Object.t.resolve (/Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:645565)
	at async /Users/ayspunde/.vscode/extensions/ms-vscode-remote.remote-ssh-0.76.1/out/extension.js:1:722872
[15:35:34.661] ------




[15:35:41.330] Password dialog canceled
[15:35:41.330] "install" terminal command canceled
[15:35:41.331] > Connection timed out during banner exchange
> Connection to UNKNOWN port 65535 timed out

**Comment by roblourens** [comment]
Please share the full log. Does it work if you set `"remote.SSH.useLocalServer": true`?

**Comment by ayspunde** [comment]
Cluster maintenance day, will get back tomorrow

**Comment by ayspunde** [comment]
1. setting "remote.SSH.useLocalServer": true
let me jump to load balancer ("cori" in "ssh -A -J cori,cmom05 nid12345", one step further than beforehand), but still failed on the next step (jumping to a cluster node))
3. attached log files
[extension_host.log](https://github.com/microsoft/vscode-remote-release/files/8319175/extension_host.log)
[stdout.log](https://github.com/microsoft/vscode-remote-release/files/8319176/stdout.log)
[window.log](https://github.com/microsoft/vscode-remote-release/files/8319177/window.log)

**Comment by roblourens** [comment]
With useLocalServer, I have no clue what caused this error, but the ssh command died.

```
[13:32:07.521] stderr> -bash: line 1: $'SSH-2.0-OpenSSH_8.6\r': command not found
[13:32:41.367] stderr> Connection timed out during banner exchange
[13:32:41.367] stderr> Connection to UNKNOWN port 65535 timed out
[13:32:41.367] > local-server-1> ssh child died, shutting down
```

Without, we were able to show the password prompt but then the ssh command died 1s later, and I don't know why:

```
[15:34:34.675] Showing password prompt
[15:35:34.611] "install" terminal command done
```

I guess this is relevant but don't know what causes it

```
 > Connection timed out during banner exchange
> Connection to UNKNOWN port 65535 timed out
```
